wntr/tests/test_morph.py

```diff
@@ -315,6 +315,29 @@ def test_skeletonize_with_controls(self):
         self.assertEqual(skel_wn.num_nodes, wn.num_nodes - 17)
         self.assertEqual(skel_wn.num_links, wn.num_links - 22)
 
+    def test_skeletonize_with_excluding_nodes_and_pipes(self):
+
+        inp_file = join(datadir, "skeletonize.inp")
+        wn = wntr.network.WaterNetworkModel(inp_file)
+
+        skel_wn = wntr.morph.skeletonize(wn, 12.0 * 0.0254, use_epanet=False, nodes_to_exclude=["13"], pipes_to_exclude=["60"])
+
+        self.assertEqual(skel_wn.num_nodes, wn.num_nodes - 17)
+        self.assertEqual(skel_wn.num_links, wn.num_links - 22)
+
+        wn = wntr.network.WaterNetworkModel(inp_file)
+
+        # Change link 60 and 11 diameter to > 12, should get some results as above
+        link = wn.get_link("60")
+        link.diameter = 16 * 0.0254
+        link = wn.get_link("11")
+        link.diameter = 16 * 0.0254
+
+        skel_wn = wntr.morph.skeletonize(wn, 12.0 * 0.0254, use_epanet=False)
+
+        self.assertEqual(skel_wn.num_nodes, wn.num_nodes - 17)
+        self.assertEqual(skel_wn.num_links, wn.num_links - 22)
+
     def test_series_merge_properties(self):
```

Comment:
This test would be stronger by asserting that pipe 60 and junction 13 are in skel_wn.

Reply:
@frank-hulo Will you be able to address the review comments?  If not, we can make the minor updates in house.

Reply:
@kaklise I could finish it within the next two weeks.

Reply:
@frank-hulo We'd like to get this PR merged into main.  Let me know if you can address the comments.  I can finish it up next week if needed.